src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -349,21 +289,24 @@ get_common_simd_info (MonoClass *vector_klass, MonoMethodSignature *csignature,
 }
 
 static void
-emit_common_simd_epilogue (TransformData *td, MonoClass *vector_klass, MonoMethodSignature *csignature, int vector_size)
+emit_common_simd_epilogue (TransformData *td, MonoClass *vector_klass, MonoMethodSignature *csignature, int vector_size, gboolean allow_void)
 {
 	td->sp -= csignature->param_count;
 	for (int i = 0; i < csignature->param_count; i++)
 		td->last_ins->sregs [i] = td->sp [i].local;
 
-	g_assert (csignature->ret->type != MONO_TYPE_VOID);
 	int ret_mt = mono_mint_type (csignature->ret);
-	if (ret_mt == MINT_TYPE_VT) {
+	if (csignature->ret->type == MONO_TYPE_VOID) {
+		g_assert (allow_void);
+		interp_ins_set_dreg (td->last_ins, 0);
```

Comment:
`0` can be a valid local, you shouldn't set it as dreg for this instruction since it will screw up the optimizations for the local. If this instruction needs to have a dreg, that is never really used, you can allocate a new one that is also pop'ed from the stack so it is never used:
```
push_simple_type (td, STACK_TYPE_I4);
td->sp--;
dreg = td->sp [0].local;
```